### PR TITLE
[opt](compound) Optimize by deleting the compound expr after obtaining the final result

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -489,11 +489,20 @@ Status SegmentIterator::_get_row_ranges_by_column_conditions() {
...
+            for (auto it = _remaining_conjunct_roots.begin();
+                 it != _remaining_conjunct_roots.end();) {
                 _pred_except_leafnode_of_andnode_evaluate_result.clear();
-                auto res = _execute_predicates_except_leafnode_of_andnode(expr);
+                auto res = _execute_predicates_except_leafnode_of_andnode(*it);
                 if (res.ok() && _pred_except_leafnode_of_andnode_evaluate_result.size() == 1) {
                     _row_bitmap &= _pred_except_leafnode_of_andnode_evaluate_result[0];
+                    // Delete expr after it obtains the final result.
+                    {
+                        std::erase_if(_common_expr_ctxs_push_down,
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:500]: if pred1 ok, pred 2 not ok, is there any side effect by erasing it from remaining roots?

Comment (zzzxl1993) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:500]: Only if all the preds in the COMPOUND expr are correctly evaluated and the overall result is obtained, can this expr be deleted.

**Comment by zzzxl1993** [comment]
run buildall

**Review by qidaye — DISMISSED** [review]
LGTM

**Comment by xiaokang** [comment]
add testcase

**Comment by zzzxl1993** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**